src/lint.ts

```diff
@@ -273,9 +284,20 @@ const lintPlugin = ViewPlugin.fromClass(class {
   }
 })
 
+const DefaultDelay = 750
+
+const lintConfig = Facet.define<LintConfig, Required<LintConfig>>({
```

Comment:
Is there a reason to use a separate facet here, rather than including this option in `lintSource` (possibly renaming it to `lintConfig`)?

Reply:
I think it would be fine to combine this one option into `lintSource` if you'd prefer that, and looking at it now, I agree that it doesn't seem correct to have `delay` in both facets.

For context, there are two things we're trying to achieve:

1. we'd like to suppress all tooltips for all linter markers in the editor content but keep the gutter tooltips
2. for errors, we want markers in both the gutter and editor content but for warnings we just want gutter markers.

We can achieve the first with this `showTooltips` option and we can achieve the second simply but somewhat hackily by styling warning markers in the content to be invisible.

I extracted this PR from a larger set of new config options I was experimenting with that enable and disable decorations both at the severity level and independently in the gutter and editor content, which seems to me like a different concern to a list of sources of diagnostics.

Reply:
How about allowing `showTooltips` to be either a boolean or a `(tooltip: Tooltip, type: "gutter" | "code") => boolean` predicate? Hiding these with CSS seems a bit wasteful.

Reply:
(Or actually the boolean part wouldn't make much sense, just a predicate would work well I suppose.)

Reply:
That would work for us for tooltips. I think having the array of `Diagnostic`s available to the predicate would also be helpful: it would allow hiding of the tooltip based on the severity of the diagnostics. It could go further by instead having a diagnostics filter function; I can imagine we might decide we'd want to filter out warnings from the tooltip if there the were errors as well. Something like  `(diagnostics: Diagnostic[], type: "gutter" | "code") => Diagnostic[]`, called `tooltipDiagnosticFilter` maybe?

A similar `showMarkers` predicate for markers would be ideal and remove the need to hide warning markers via CSS.

Reply:
Oh, right, my `Tooltip` argument there was nonsense, we want to pass diagnostics, not tooltips. Your type sounds good.

Reply:
I've implemented `tooltipDiagnosticFilter` and `markerDiagnosticFilter` config options. I've also merged the lint source and config facets, but I'm really not convinced by how I've implemented that. I wasn't sure what the best way to combine values would be, given that sources should be additive and configs should be merged.

One other contextual thing I've forgotten to mention that is that the "we" above is Overleaf, who I now work for.